scripts/bin/auto-mount

```diff
@@ -1,17 +1,80 @@
-#! /bin/sh
-
-sleep 5
-
-$SNAP/usr/local/bin/udisksctl monitor | grep --line-buffered " Device:" | while read -r -- device; do
-
-  echo "$SNAP_NAME.auto-mount found ${device}"
-  if echo $device| grep -q "[0-9]$"; then
-    DEV="$(echo $device | sed 's/^Device: //')"
-  fi
-  if [ -n "$DEV" ]; then
-    RES=`$SNAP/usr/local/bin/udisksctl mount --no-user-interaction --block-device "$DEV"`
-    # process returned value into the mount point string
-    RES2=`echo $RES | cut -f4 -d\  | cut -f1 -d.`
-    echo $RES2 > $WATCHED
-  fi
-done
+#!/bin/sh
+
+wait_for_fifo() {
+    [ ! -p "$WATCH_FILE" ] || return 0
+
+    echo "Waiting for FIFO: ${WATCH_FILE}"
+    inotifywait --quiet --csv --event create "$(dirname "$WATCH_FILE")" | \
+        while IFS=, read -r _dir _ _name; do
+            [ "${_dir}/${_name}" != "$WATCH_FILE" ] || return 0
+        done
+}
+
+watch_devices() {
+    inotifywait --quiet --monitor --event "moved_to" --event "delete" /dev/disk/by-uuid | \
+        while read -r _parent _action _uuid; do
+            case "$_action" in
+                MOVED_TO) try_mount "$_parent" "$_uuid" ;;
+                *) : ;;
+            esac
+        done
+}
+
+try_mount() {
+    _parent="${1%/}"
+    _uuid="$2"
+
+    _source="${_parent}/${_uuid}"
+    _target="${MOUNT_DIR}/${_uuid}"
+
+    echo "${SNAP_NAME}.auto-mount found: ${_source}"
+
+    mkdir -p "$_target"
+    grep -q "$_target" /proc/mounts || {
+        if ! mount -o rw,sync -t "${FSTYPE:-vfat}" "$_source" "$_target" >/dev/null 2>&1; then
+            echo "WARN: Could not mount ${_source} to ${_target}"
+            rm -rf "$_target"
+            return 0
+        fi
+    }
+
+    wait_for_fifo
+    echo "$_target" >> "$WATCH_FILE"
+}
+
+cleanup () {
+    _exit_status=$?
+
+    [ -z "${CLEANUP_HAS_RUN:-}" ] || return 0
+    CLEANUP_HAS_RUN=1
+
+    sync
+
+    _uuids="$(ls -1 "$MOUNT_DIR" | tr '\n' :)"
```

Comment:
I dislike parsing `ls` but `find` here is... unpleasantly unsupported by POSIX...
I think this stays but we should remember to resolve this; mark with TODO